kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java

```diff
@@ -906,6 +908,34 @@ private InternalClusterMetrics calculateClusterMetrics(
     return result;
   }
 
+  public Mono<BrokerLogdirUpdateResult> updateBrokerLogDir(KafkaCluster cluster, Integer broker,
+                                                           BrokerLogdirUpdate brokerLogDir) {
+    return getOrCreateAdminClient(cluster)
+        .flatMap(ac -> updateBrokerLogDir(ac, brokerLogDir, broker));
+  }
 
+  private Mono<BrokerLogdirUpdateResult> updateBrokerLogDir(ExtendedAdminClient adminMono,
+                                                            BrokerLogdirUpdate b,
+                                                            Integer broker) {
+
+    Map<TopicPartitionReplica, String> req = Map.of(
+        new TopicPartitionReplica(b.getTopic(), b.getPartition(), broker),
+        b.getLogDir());
+    return Mono.just(adminMono)
+        .map(admin -> admin.getAdminClient().alterReplicaLogDirs(req))
+        .flatMap(result -> ClusterUtil.toMono(result.all()))
+        .then(Mono.just(result(BrokerLogdirUpdateResult.StatusEnum.OK, null)))
+        .onErrorResume(ApiException.class, (ApiException e) -> {
+          log.error("Error during updating log dirs", e);
+          return Mono.just(result(BrokerLogdirUpdateResult.StatusEnum.ERROR, e.getMessage()));
```

Comment:
Looks like you'll return 200 even if broker sent error. I think it's better to pass 4xx/5xx status

Reply:
Fixed, please review